index_part.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * go-leia
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package leia
+
+// IndexOption is the common type for indexing options
+// It should not be used directly
+type IndexOption interface{}
```

Comment:
The option pattern in Go usually involves the `IndexOption` being a function signature that accepts the data structure the option is applied to. I guess the way it's done here works, but isn't as extensible (because each option has to be both defined and then implemented in `fieldIndexer` as well). Whereas usually you implement the behavior in the option itself.

Reply:
changed to funcs, is better that way